src/components/Card.js

```diff
@@ -14,12 +25,19 @@ export default class Card {
     this._cardImage = this._element.querySelector("#card__image");
     this._cardTitle = this._element.querySelector("#card__title");
   }
+
   _handleLikeIcon() {
-    this._likeBtn.classList.toggle("card__like-button_active");
+    if (!this._likeBtn.classList.contains("card__like-button_active")) {
```

Comment:
Needs correction: You should change the button state only after successful api response

The code for rendering likes should look like this:

```js
// in Card class 

// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 

isLiked() {
  return this._isLiked;
}

_renderLikes() {
  if (this._isLiked) {
    // add active class to the like button
  } else {
    // remove active class from the like button
  }
}

createCard() { 
  ...
  // render initial likes, you saved them in the constructor
  this._renderLikes()
}

// index.js
if (card.isLiked()) {
  api.dislikeCard(...).then(response => card.setIsLiked(response._isLiked)).catch(() => /*handle error*/)
} else {
 // the same for adding like
}
```

Reply:
Got it.